go-controller/pkg/node/controllers/egressservice/egressservice.go

```diff
@@ -0,0 +1,927 @@
+
+	endpointSlices, err := c.endpointSliceLister.EndpointSlices(svc.Namespace).List(esLabelSelector)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	v4Endpoints := sets.New[string]()
+	v6Endpoints := sets.New[string]()
+
+	for _, eps := range endpointSlices {
+		if eps.AddressType == discoveryv1.AddressTypeFQDN {
+			continue
+		}
+
+		epsToInsert := v4Endpoints
+		if eps.AddressType == discoveryv1.AddressTypeIPv6 {
+			epsToInsert = v6Endpoints
+		}
+
+		for _, ep := range eps.Endpoints {
+			for _, ip := range ep.Addresses {
+				ipStr := utilnet.ParseIPSloppy(ip).String()
+				if !services.IsHostEndpoint(ipStr) {
+					epsToInsert.Insert(ipStr)
+				}
+			}
+		}
+	}
+
+	return v4Endpoints, v6Endpoints, nil
+}
+
+// Clears all of the SNAT rules of the service.
+func (c *Controller) clearServiceSNATRules(key string, state *svcState) error {
+	for ip := range state.v4Eps {
+		err := nodeipt.DelRules([]nodeipt.Rule{snatIPTRuleFor(key, state.v4LB, ip)})
+		if err != nil {
+			return err
+		}
+
+		state.v4Eps.Delete(ip)
+	}
+
+	for ip := range state.v6Eps {
+		err := nodeipt.DelRules([]nodeipt.Rule{snatIPTRuleFor(key, state.v6LB, ip)})
+		if err != nil {
+			return err
+		}
+
+		state.v6Eps.Delete(ip)
+	}
+
+	state.v4LB = ""
+	state.v6LB = ""
+
+	return nil
+}
+
+// Clears all of the FWMark rules of the service.
+func (c *Controller) clearServiceIPRules(state *svcState) error {
+	errorList := []error{}
+	for ip := range state.netEps {
+		family := "-4"
+		if utilnet.IsIPv6String(ip) {
+			family = "-6"
+		}
+
+		err := deleteIPRule(family, IPRulePriority, ip, state.net)
+		if err != nil {
+			errorList = append(errorList, err)
+			continue
+		}
+
+		state.netEps.Delete(ip)
+	}
+
+	return errors.NewAggregate(errorList)
+}
+
+// Clears all of the iptables rules that relate to the service and removes it from the cache.
+func (c *Controller) clearServiceRules(key string, state *svcState) error {
+	state.stale = true
+
+	err := c.clearServiceSNATRules(key, state)
+	if err != nil {
+		return err
+	}
+
+	err = c.clearServiceIPRules(state)
+	if err != nil {
+		return err
+	}
+
+	delete(c.services, key)
+	c.egressServiceQueue.Add(key)
+
+	return nil
+}
+
+// Returns true if the controller should configure the given service as an "Egress Service"
+func (c *Controller) shouldConfigureEgressSVC(svc *corev1.Service, svcHost string) bool {
+	return svcHost == c.thisNode &&
+		svc.Spec.Type == corev1.ServiceTypeLoadBalancer &&
+		len(svc.Status.LoadBalancer.Ingress) > 0
+}
+
+// Create the ip rule to use the specified routing table for the fwmark.
+// The rule is created with the same priority as the fwmark, and if a rule
+// exists with that priority it deletes it beforehand.
+func createIPRule(family string, priority int32, src, table string) error {
+	prio := fmt.Sprintf("%d", priority)
+	stdout, stderr, err := util.RunIP(family, "rule", "add", "prio", prio, "from", src, "table", table)
+	if err != nil && !strings.Contains(stderr, "File exists") {
```

Comment:
I think this is more readable and reasonable enough for this usecase. do you strongly prefer I check how it could be done using codes (could be quite complicated since I think util.RunIP abstracts it)?

Reply:
I do not have a strong preference here but I think that, in general, the error code is less likely to change than the stderr output. 
Example of error code handling in k8s: https://github.com/kubernetes/kubernetes/blob/master/pkg/util/iptables/iptables.go#L260

We can leave it as is for now.